dependencies.py

```diff
@@ -31,21 +31,20 @@ def extra_requirements() -> typing.Dict[str, typing.List[str]]:
     #     - We have a copy of these in extras-requirements.txt. If you modify these, make sure to change it
     #       there as well
     extras_require = {
-        # from 1.17.107 boto3 requires botocore>=1.20.107,<1.21.0 which
-        # conflicts with s3fs 2021.8.1 that has aiobotocore~=1.4.0
-        # which so far (1.4.1) has botocore>=1.20.106,<1.20.107
-        # boto3 1.17.106 has botocore>=1.20.106,<1.21.0, so we must add botocore explicitly
+        # last version that supports python 3.7: fsspec: 2023.1.0, aiobotocore: 2.4.2
+        # selecting fsspec 2023.1.0 and same version for the s3fs and gcsfs implementations
+        # s3fs 2023.1.0 requires aiobotocore 2.4.2 which requires botocore 1.27.59
+        # requesting boto3 1.24.59, the only version that requires botocore 1.27.59
         "s3": [
-            "boto3~=1.9, <1.17.107",
-            "botocore>=1.20.106,<1.20.107",
-            "aiobotocore~=1.4.0",
-            "s3fs~=2021.8.1",
+            "boto3~=1.24.59",
+            "aiobotocore~=2.4.2",
+            "s3fs~=2023.1.0",
```

Comment:
The comments imply that we lock all these versions, but the use of `~=` means that we allow them all to take a later patch version.

Reply:
changed the wording in the comment to make the use of ~= clear

Reply:
Keep in mind that I was referring to boto dependencies as well. I.e., we're actually requesting boto3 1.24.59 or a later patch version.

Reply:
in practice what i need is for aiobotocore and boto3 to agree on the botocore version. in theory these two projects may release a new patch version which bump botocore - and I guess we would like to take these patches..